processing_provider/datagen.py

```diff
@@ -0,0 +1,184 @@
+        should provide a basic description about what the algorithm does and the
+        parameters and outputs associated with it..
+        """
+        return self.tr("Preprocess layer into predictable tiles")
+
+    def initAlgorithm(self, config=None):
+        """
+        Here we define the inputs and output of the algorithm, along
+        with some other properties.
+        """
+
+        # We add the input vector features source. It can have any kind of
+        # geometry.
+        self.addParameter(
+            QgsProcessingParameterRasterLayer(
+                self.INPUT,
+                self.tr('Input layer')
+            )
+        )
+
+        # We add a feature sink in which to store our processed features (this
+        # usually takes the form of a newly created vector layer when the
+        # algorithm is run in QGIS).
+        self.addParameter(
+            QgsProcessingParameterString(
+                self.DATASET,
+                self.tr('Dataset name')
+            )
+        )
+
+        self.addParameter(
+            QgsProcessingParameterNumber(
+                self.SHAPE,
+                self.tr('Number of pixel for the side of tiles'),
+                type = QgsProcessingParameterNumber.Integer,
+                defaultValue = 512,
+                minValue = 16
+            )
+        )
+
+        self.addParameter(
+            QgsProcessingParameterFolderDestination(
+                self.OUTPUT,
+                self.tr('Output folder')
+            )
+        )
+
+    def processAlgorithm(self, parameters, context, feedback):
+        """
+        Here is where the processing itself takes place.
+        """
+
+        raster_in = self.parameterAsRasterLayer(
+            parameters,
+            self.INPUT,
+            context
+        )
+
+        dest_path = self.parameterAsString(
+            parameters,
+            self.OUTPUT,
+            context
+        )
+
+        dataset = self.parameterAsString(
+            parameters,
+            self.DATASET,
+            context
+        )
+
+        shape = self.parameterAsInt(
+            parameters,
+            self.SHAPE,
+            context
+        )
+
+        path=''
+        for i in [dest_path, dataset, 'input','testing','images']:
+            path = os.path.join(path, i)
+            if not os.path.exists(path):
+                os.mkdir(path)
+
+        for file in os.listdir(path):
+            os.remove(os.path.join(path,file))
+
+        shutil.copy( raster_in.source(), os.path.join( path, os.path.basename( raster_in.source())))
+
+        output_folder = os.path.join(dest_path, dataset, 'preprocessed', str(shape), 'testing', 'images')
+        shutil.rmtree(os.path.join(dest_path, dataset, 'preprocessed', str(shape)))
+        
+        cmd = ['deepo', 'datagen', '-D', dataset, '-s', str(shape), '-P', dest_path, '-T', '1']
+        subprocess.run(cmd)
```

Comment:
I'm kind of puzzled with the call to `subprocess`. I'm sure we could use the `deeposlandia` API; let's say it's OK there, however it deserves a special issue.